src/ctia/stores/es/crud.clj

```diff
@@ -37,47 +47,130 @@
            (map #(dissoc % :_id :_index :_type))
            (map coerce!)))))
 
+(s/defn allowed-group? :- s/Bool
+  [doc ident]
+  (boolean
+   (and (seq (:groups ident))
+        (seq (:groups doc))
+        (seq (clojure.set/intersection
+              (set (:groups ident))
+              (set (:groups doc)))))))
+
+(s/defn owner? :- s/Bool
+  [doc ident]
+  (boolean
+   (and (= (:owner doc)
+           (:login ident))
+        (allowed-group? doc ident))))
```

Comment:
I agree that this can be somewhat misleading, 
depending on the authorization module being used, we cannot guarantee user "uniqueness" across all organizations, in our current situation there might be 2 owners "foo" on 2 different orgs, thus we need to also check that the user matches a group.

I think we might also want to enforce the auth modules to prefix the owner string with their provider ids, or the origin group.